server/src/main/java/org/opensearch/search/pipeline/SearchPipelineService.java

```diff
@@ -354,13 +358,28 @@ public PipelinedRequest resolvePipeline(SearchRequest searchRequest) throws Exce
             } catch (Exception e) {
                 throw new SearchPipelineProcessingException(e);
             }
-        } else if (searchRequest.pipeline() != null) {
-            String pipelineId = searchRequest.pipeline();
-            PipelineHolder pipelineHolder = pipelines.get(pipelineId);
-            if (pipelineHolder == null) {
-                throw new IllegalArgumentException("Pipeline " + pipelineId + " is not defined");
+        } else {
+            String pipelineId = NOOP_PIPELINE_ID;
+            if (searchRequest.pipeline() != null) {
+                // Named pipeline specified for the request
+                pipelineId = searchRequest.pipeline();
+            } else if (searchRequest.indices() != null && searchRequest.indices().length == 1) {
+                // Check for index default pipeline
+                IndexMetadata indexMetadata = state.metadata().index(searchRequest.indices()[0]);
+                if (indexMetadata != null) {
+                    Settings indexSettings = indexMetadata.getSettings();
+                    if (IndexSettings.DEFAULT_SEARCH_PIPELINE.exists(indexSettings)) {
```

Comment:
Use `indexSettings.getDefaultSearchPipeline()` ?

Reply:
The problem is that `indexSettings` is not an instance of `IndexSettings`. 

I looked through it and there's no connection between `IndexMetadata` and `IndexSettings` right now.